src/app/modules/submission/submission.service.ts

```diff
@@ -203,3 +207,58 @@ export const findSubmissionById = (
     return okAsync(submission)
   })
 }
+
+/**
+ * @param pendingSubmissionId the id of the pending submission to confirm
+ * @param session (optional) the mongoose transaction session to be used, if any
```

Comment:
is there a case where this will be done outside of a transaction session?

Reply:
I think we should enforce it. It can be a costly mistake, and an evasive bug.